src/python.cpp

```diff
@@ -2207,3 +2207,91 @@ SEXP py_eval_impl(const std::string& code, bool convert = true) {
   return result;
 
 }
+
+// [[Rcpp::export]]
+SEXP py_convert_pandas_df(PyObjectRef obj) {
+  
+  // pd.DataFrame.items() iterates over (column name, Series) pairs
+  PyObjectPtr items(PyObject_CallMethod(obj, "items", NULL));
```

Comment:
Can we double-check that this does indeed return an iterable Python object? (Ensuring it has a `next` or `__next__` attribute should be sufficient)

Reply:
done